### DISCUSSION TITLE
Quit not actually quits, how to use it?

### DISCUSSION THREAD

**Discussion by fnaj77** [janet-lang/janet#1237] [Q&A] [answered]
Hi guys,
I'm having some trouble figuring out how to use quit.
The following script doesn't work as I would expect, 'cause the string `actually not` is printed on screen.

```
(import spork)

(defn main
  [ & args ]
  (if (not (spork/sh/exists? "i-do-not.exist"))
    (do
      (print "file does not exist, I'm exiting")
      (quit)))
  (print "actually not"))

```
I read in the [guide](https://janetdocs.com/quit) that quit:
>... Does not always exit the application ...

How can I force stop the execution?
I'm running the latest janet version compiled from source on a ubuntu 22.10 machine.

Thanks.

**Comment by ianthehenry** [comment, marked as answer]
Use `os/exit` or remove the `main` definition. `quit` only really makes sense as a top-level statement — it stops evaluating the rest of the file, but doesn’t exit the process.

**Reply by fnaj77** [reply]
Hi Ian,
thank you very much for your help!
